### ISSUE TITLE
dataset documentation does not match behavior

### ISSUE THREAD

**Issue by zenogantner** [hpclab/rankeval#4]
I have a 92921 line input file in LibSVM/RankLib format, with 2155 query IDs:
```
> wc -l output.libsvm
92921 output.libsvm
> cut -f 2 -d ' ' output.libsvm | sort | uniq | wc -l
2155
```

```
dataset = Dataset.load("output.libsvm")
print(dataset)
print("n_queries: %s" % dataset.n_queries)
print("len(query_ids): %s" % len(dataset.query_ids))
```
Expected output (according to documentation):
```
Dataset (92921, 6)
n_queries: 2155
len(query_ids): 92921
```

Actual output:
```
Dataset (92921, 6)
n_queries: 2155
len(query_ids): 2156
```

Documentation snippet in https://github.com/hpclab/rankeval/blob/master/rankeval/dataset/dataset.py I am referring to:
```
query_ids : numpy 1d array of int
        It is a ndarray of shape(nsamples,)
```

I think the reason why it works not as expected can be found in the constructor:
```
if len(query_ids) == X.shape[0]:
```

I guess this is done for easier by-query access.
However, it does not match the documented behavior.
Either the implementation or the documentation should be changed.

Note also that the logic here does not work in the case n_queries == n_instances.

**Comment by zenogantner** [comment]
I would like to have the "raw" query IDs per instance row, because I want to split my datasets ...

**Comment by strani** [comment]
I pushed a new version of rankeval, with tag 0.6, that introduce quite a lot of novelties on the dataset class (as well as other fixes and improvements), including the features you are searching for. The documentation in the Dataset class has been changed accordingly. The problem with the constructor logic is not still there, but I would also like to point out that in LtR it is not feasible to have a dataset with n_queries == n_instances.

Finally, a added a novel notebook explaining how to use the several features provided by the Dataset class and the meaning of the low-level information stored there.

Feedbacks are welcomed.
